tv_apps/smart-system/js/accessibility.js

```diff
@@ -558,30 +551,27 @@
         this.cancel();
       }
 
-      var sentence = this.buildUtterance(aData);
-      if (!sentence) {
-        if (aCallback) {
-          aCallback();
-        }
-        return;
-      }
-
-      var utterance = new this.utterance(sentence);
-      utterance.volume = aVolume;
-      utterance.rate = aRate;
-      utterance.addEventListener('end', function() {
-        if (this.captions) {
-          this.hideSpeech();
-        }
-        if (aCallback) {
-          aCallback();
+      return this.buildUtterance(aData).then(sentence => {
+        if (!sentence) {
+          return Promise.resolve();
```

Comment:
I guess it's just an leftover. How about try removing `aCallback`, this may need to change some parameters like `Accessibility.announceScreenReader` and `Accessibility.speak`.